src/main/java/org/adridadou/ethereum/propeller/EthereumProxy.java

```diff
@@ -158,11 +158,21 @@ SmartContractByteCode getCode(EthAddress address) {
                 .flatMap(receipt -> {
                     List<EventData> events = receipt.events;
                     return Observable.fromIterable(events.stream().filter(eventDefinition::match)
-                            .map(data -> new EventInfo<>(receipt.hash, eventDefinition.parseEvent(data)))
+                            .map(data -> new EventInfo<>(receipt.hash, parseEventSafely(eventDefinition, data)))
+                            .filter(event -> event != null)
                             .collect(Collectors.toList()));
                 });
     }
 
+    private <T> T parseEventSafely(SolidityEvent<T> eventDefinition, EventData data) {
+        try {
+          return eventDefinition.parseEvent(data);
+        } catch (Exception e) {
+          logger.warn("failed parsing event for ABI: " + eventDefinition.getDescription());
+          return null;
```

Comment:
I think it's problematic to return a null here. If I'm not mistaken Observable assumes non null values